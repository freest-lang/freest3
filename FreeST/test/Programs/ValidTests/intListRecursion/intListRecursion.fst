
mkList : Int -> [Int]
mkList x = if x == 0 then [] else x :: mkList (x-1)

getMultiples : Int -> [Int] -> [Int]
getMultiples n list =
    case list of {
        [] -> [],
        (x::xs) -> 
            if mod x n == 0 
                then x :: (getMultiples n xs)
                else getMultiples n xs
    }

main : [Int]
main = getMultiples 2 (getMultiples 9 (mkList 1000))