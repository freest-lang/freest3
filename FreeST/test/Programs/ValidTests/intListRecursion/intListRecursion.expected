[990,972,954,936,918,900,882,864,846,828,810,792,774,756,738,720,702,684,666,648,630,612,594,576,558,540,522,504,486,468,450,432,414,396,378,360,342,324,306,288,270,252,234,216,198,180,162,144,126,108,90,72,54,36,18]